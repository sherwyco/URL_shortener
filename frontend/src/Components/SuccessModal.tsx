import { useState } from 'react';
import Box from '@mui/material/Box';
import Typography from '@mui/material/Typography';
import { TextField } from '@mui/material';
import Modal from '@mui/material/Modal';

const style = {
    position: 'absolute',
    top: '50%',
    left: '50%',
    transform: 'translate(-50%, -50%)',
    width: 500,
    bgcolor: 'background.paper',
    border: '2px solid #000',
    boxShadow: 24,
    p: 4,
};

interface ModalPropTypes {
    shortUrl: string;
}

function SuccessModal(props: ModalPropTypes) {
    const { shortUrl } = props;
    const [open, setOpen] = useState(true);
    const handleClose = () => setOpen(false);

    return (
        <div>
            <Modal
                open={open}
                onClose={handleClose}
                aria-labelledby="modal-modal-title"
                aria-describedby="modal-modal-description"
            >
                <Box sx={style}>
                    <h3>Success!</h3>
                    <Typography id="modal-modal-description" >
                        Your short url:
                    </Typography>
                    <TextField
                        type='text'
                        defaultValue={shortUrl}
                        variant='outlined'
                        inputProps={
                            { readOnly: true, }
                        }
                        fullWidth
                    />
                </Box>
            </Modal>
        </div >
    );
}
export default SuccessModal;