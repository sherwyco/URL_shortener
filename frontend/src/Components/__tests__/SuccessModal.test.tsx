import { render } from '@testing-library/react';
import SuccessModal from '../SuccessModal';

it('matches component snapshot', () => {
    const { baseElement } = render(<SuccessModal shortUrl='1234567' />)
    expect(baseElement).toMatchSnapshot();
})