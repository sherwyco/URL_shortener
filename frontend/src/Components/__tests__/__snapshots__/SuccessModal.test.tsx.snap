// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches component snapshot 1`] = `
<body
  style="padding-right: 1024px; overflow: hidden;"
>
  <div
    aria-hidden="true"
  >
    <div />
  </div>
  <div
    aria-describedby="modal-modal-description"
    aria-labelledby="modal-modal-title"
    class="MuiModal-root css-79ws1d-MuiModal-root"
    role="presentation"
  >
    <div
      aria-hidden="true"
      class="MuiBackdrop-root MuiModal-backdrop css-i9fmh8-MuiBackdrop-root-MuiModal-backdrop"
      style="opacity: 1; webkit-transition: opacity 225ms cubic-bezier(0.4, 0, 0.2, 1) 0ms; transition: opacity 225ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;"
    />
    <div
      data-testid="sentinelStart"
      tabindex="0"
    />
    <div
      class="MuiBox-root css-s0w8ce"
      tabindex="-1"
    >
      <h3>
        Success!
      </h3>
      <p
        class="MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root"
        id="modal-modal-description"
      >
        Your short url:
      </p>
      <div
        class="MuiFormControl-root MuiFormControl-fullWidth MuiTextField-root css-wb57ya-MuiFormControl-root-MuiTextField-root"
      >
        <div
          class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-fullWidth MuiInputBase-formControl css-md26zr-MuiInputBase-root-MuiOutlinedInput-root"
        >
          <input
            aria-invalid="false"
            class="MuiInputBase-input MuiOutlinedInput-input css-1t8l2tu-MuiInputBase-input-MuiOutlinedInput-input"
            id=":r0:"
            readonly=""
            type="text"
            value="1234567"
          />
          <fieldset
            aria-hidden="true"
            class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
          >
            <legend
              class="css-ihdtdm"
            >
              <span
                class="notranslate"
              >
                â€‹
              </span>
            </legend>
          </fieldset>
        </div>
      </div>
    </div>
    <div
      data-testid="sentinelEnd"
      tabindex="0"
    />
  </div>
</body>
`;
